flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvRowDeserializationSchema.java

```diff
@@ -259,9 +264,19 @@ private static RuntimeConverter assembleRowRuntimeConverter(
             for (int i = 0; i < Math.min(rowArity, nodeSize); i++) {
                 // Jackson only supports mapping by name in the first level
                 if (isTopLevel) {
-                    row.setField(i, fieldConverters[i].convert(node.get(fieldNames[i])));
+                    try {
+                        row.setField(i, fieldConverters[i].convert(node.get(fieldNames[i])));
+                    } catch (Throwable t) {
+                        throw new IllegalStateException(
+                                String.format(errMessage, fieldNames[i]), t);
+                    }
                 } else {
-                    row.setField(i, fieldConverters[i].convert(node.get(i)));
+                    try {
+                        row.setField(i, fieldConverters[i].convert(node.get(i)));
+                    } catch (Throwable t) {
+                        throw new IllegalStateException(
+                                String.format(errMessage, fieldNames[i]), t);
+                    }
```

Comment:
It seems it's a little duplicate here. Could you add a function to reuse the logic?

Would be better to use `RuntimeException` to align with others.